homeassistant/components/fujitsu_hvac/climate.py

```diff
@@ -0,0 +1,188 @@
+"""Support for Fujitsu HVAC devices that use the Ayla Iot platform."""
+from typing import Any
+
+from ayla_iot_unofficial.fujitsu_hvac import Capability, FujitsuHVAC
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_HALVES, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    DOMAIN,
+    FUJI_TO_HA_FAN,
+    FUJI_TO_HA_HVAC,
+    FUJI_TO_HA_SWING,
+    HA_TO_FUJI_FAN,
+    HA_TO_FUJI_HVAC,
+    HA_TO_FUJI_SWING,
+)
+from .coordinator import FujitsuHVACCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up one Fujitsu HVAC device."""
+    api = hass.data[DOMAIN][entry.entry_id]
+    coordinator = FujitsuHVACCoordinator(hass, api)
+    await coordinator.async_config_entry_first_refresh()
+
+    async_add_entities(
+        [FujitsuHVACDevice(coordinator, dev) for dev in coordinator.data.values()]
+    )
+
+
+class FujitsuHVACDevice(CoordinatorEntity, ClimateEntity):
+    """Represent a Fujitsu HVAC device."""
+
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_precision = PRECISION_HALVES
+    _attr_target_temperature_step = 0.5
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    _enable_turn_on_off_backwards_compatibility: bool = False
+
+    def __init__(self, coordinator: FujitsuHVACCoordinator, dev: FujitsuHVAC) -> None:
+        """Set up static attributes."""
+        super().__init__(coordinator, context=dev.device_serial_number)
```

Comment:
```suggestion
        super().__init__(coordinator)
```

Not sure what context actually does here, but I don't think its needed

Reply:
Context is what is returned by async_contexts in the coordinator. This allows to only fetch the data for devices that are actually registered with the coordinator.